common/v2/services/RatesProvider.tsx

```diff
@@ -36,32 +84,64 @@ export function RatesProvider({ children }: { children: React.ReactNode }) {
     // Save settings rates again when the assets change.
     setIsSettingsInitialized(false);
   }, [Object.keys(rates)]);
-  const currentTickers = assetTickers();
+
+  useEffect(() => {
+    if (!worker) return;
+    worker.start();
+  }, [worker]);
+
+  const currentAssetIDs = assetIDs();
   useEffect(() => {
     // The cryptocompare api that our proxie uses fails gracefully and will return a conversion rate
     // even if some are tickers are invalid (e.g WETH, GoerliETH etc.)
+    fetchAssetMappingList().then((coinMappingObj: AssetMappingListObject) => {
+      const formattedCoinGeckoIds = pullCoinGeckoIDs(coinMappingObj, currentAssetIDs as string[]);
 
-    const worker = new PollingService(
-      buildQueryUrl(currentTickers, DEFAULT_FIAT_PAIRS), // @TODO: figure out how to handle the conversion more elegantly then `DEFAULT_FIAT_RATE`
-      POLLING_INTERRVAL,
-      (data: IRates) => setRates(data),
-      err => console.debug('[RatesProvider]', err)
-    );
+      const createWorker = () => {
+        setWorker(
+          new PollingService(
+            buildTokenQueryUrl(formattedCoinGeckoIds, DEFAULT_FIAT_PAIRS), // @TODO: figure out how to handle the conversion more elegantly then `DEFAULT_FIAT_RATE`
+            POLLING_INTERRVAL,
+            (data: IRates) =>
+              setRates({ ...rates, ...destructureCoinGeckoIds(data, coinMappingObj) }),
+            err => {
+              console.debug('[RatesProvider]', err);
+              terminateWorker();
+            }
+          )
+        );
+      };
 
-    const terminateWorker = () => {
-      worker.stop();
-      worker.close();
-    };
+      if (!worker) {
+        createWorker();
+        return;
+      }
 
-    worker.start();
-    return terminateWorker; // make sure we terminate the previous worker on teardown.
-  }, [rawAccounts, currentTickers.length]); // only update if an account has been added or removed from LocalStorage or if assets have been updated.
+      const terminateWorker = () => {
+        worker.stop();
+        worker.close();
+      };
+
+      // Stop old worker
+      terminateWorker();
+      // Create new worker.
+      createWorker();
```

Comment:
The rules for stopping and starting the worker are quite confusing.
Can you share the conditions you identified?